### PR TITLE
City update

### PR DESCRIPTION
**Pull request by rovenr12**

### CODE REVIEW

File: src/core/game/Board.java [hunk 1]
```diff
@@ -444,14 +447,34 @@ public int getCityIdAt(int x, int y)
...
+        LinkedList<Integer> cities = new LinkedList<>(t.getCitiesID());
+        cities.remove((Integer)t.getCapitalID());
+        boolean ownCapital = capital.getTribeId() == t.getTribeId();
         City c;
         if(ter == Types.TERRAIN.VILLAGE)
         {
             //Not a city. Needs to be created, assigned and its border calculated.
             c = new City(x, y, t.getTribeId());
+            // Move the unit from one city to village. Rank: capital -> cities -> None
+            if(ownCapital && capital.getUnitsID().size() > 0){
```
Comment (GAIGResearch) [inline src/core/game/Board.java]: ownCapital would always be false here, right? No villages can be converted into capitals.

Or is this meant to be true if the tribe _owns_ the capital? In that case, Tribe.controlsCapital() gives you that value. Same for preOwnCapital below in line 491

Comment (rovenr12) [reply src/core/game/Board.java]: This function is not available in my version. Could I merge and then change to it?

Comment (GAIGResearch) [reply src/core/game/Board.java]: Would it be possible for you to merge the current master into your branch? That should bring this function and all the other changes before we merge back to master.

Comment (rovenr12) [reply src/core/game/Board.java]: there is conflict, I cannot merge it

Comment (GAIGResearch) [reply src/core/game/Board.java]: Okay, let's merge and then we make the change. There are conflicts city_update -> master, I'll try to solve them now and I'll get back to you when ready.

**Pull request merged**